doc/design/couler_sqlflow.md

```diff
@@ -0,0 +1,49 @@
+# Couler on SQLFlow
+
+## Motivations
+
+The purpose of Couler is to provide a joyful experience of writing workflows runnable on professional engines like Argo. SQLFlow translates a SQL program into a Python machine-learning program in Python.
+This architle introduces how does SQLFlow translate a SQLFlow program into a Couler workflow program.
+
+## SQLFlow and Couler
+
+The Couler core package implemented many functions like:
+1. `couler.run_container(docker_image, cmd, args)` starts a Docker contanier.
+1. `couler.run_python(python_func_name, docker_image="python:3.6")` runs a Python function in the given Docker image.
+
+For SQLFlow, we would like to implement multiple Python function to Train/Predict TensorFlow/XGBoost/... model:
+
+``` python
+couler.python_run(python_func_name="xgboost.train", ir=IR)
```

Comment:
Only the codegen_couler.go has the chance to take an IR (intermediate representation). A Couler function doesn't have the chance.